WordPress/Classes/Models/ReaderPost.m

```diff
@@ -505,4 +506,17 @@ - (NSDictionary *)railcarDictionary
     return nil;
 }
 
+- (void) didSave {
+    [super didSave];
+
+    // When saving a post, changes are not reflected in Reader Discover because of the usage of
+    // ReaderCard as the Core Data object. By "faking" this change we make sure Reader Discover
+    // reflects a change on a post.
+    // We can confidently set `topics` to NULL given a card represents a post OR a list of topics.
+    if (self.card.count > 0) {
+        self.card.allObjects[0].topics = NULL;
+        [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
```

Comment:
I personally don't like to have a reference to Core Data inside this model. However, it would be that or a reference to the service. Either feels wrong to me.

Anyway, it's that or creating a lot of code to make `NSFetchedResultsController` to observe changes on a child. 😬